File: src/market-tracker-webapi/Application/Domain/Schemas/List/ListEntry.cs
using market_tracker_webapi.Application.Domain.Schemas.Market.Inventory.Product;
using market_tracker_webapi.Application.Domain.Schemas.Market.Retail.Shop;

namespace market_tracker_webapi.Application.Domain.Schemas.List;

public record ListEntry(ListEntryId Id, Product Product, StoreItem? Store, int Quantity);

public record ListEntryOffer(ProductOffer ProductOffer, int Quantity);

public record ListEntryId(
    string Value
);

File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/domain/model/list/listEntry/ListEntry.kt
package pt.isel.markettracker.domain.model.list.listEntry

import pt.isel.markettracker.domain.model.market.Store
import pt.isel.markettracker.domain.model.market.inventory.product.ProductOffer

data class ListEntry(
    val id: String,
    val productId: String,
    val storeId: Store?,
    val quantity: Int,
)

data class ListEntryOffer(val id: String, val productOffer: ProductOffer, val quantity: Int)